OneCommunityGlobal/HGNRest
# Xiao fix x hours in one week badges

# Description
![Screenshot 2024-03-19 at 14 46 15](https://github.com/OneCommunityGlobal/HGNRest/assets/113070229/dd7c6827-e550-426a-8cbe-46bc09cfd504)

## Related PRS (if any):
N/A

## Main changes explained:
- Split the existing XHoursForXWeeks function into two separate entities
- Create a function to allocate badges for X hours in 1 week.

## How to test:
1. Check into the current branch
2. Comment out the test code in the `userProfileJobs.js` under `HGNREST/src/cronjobs/` so that it can run every minute, not every Sunday night. 
![Screenshot 2024-03-19 at 14 53 56](https://github.com/OneCommunityGlobal/HGNRest/assets/113070229/a2ac2f15-436c-4e57-841f-fa636321e9d7)
3. Also, change the query in the `awardNewBadges` function in `userHelper.js` under `HGNREST/src/helpers/` from all active users to whatever user you own so that the test will not affect all users. 
![Screenshot 2024-03-19 at 15 11 50](https://github.com/OneCommunityGlobal/HGNRest/assets/113070229/62f5cfc6-dd69-497f-9019-76fb51f115f7)
4. Then go to MongoDB Compass and search for the user you entered in the previous step in the `userProfile` table. Modify the `lastWeekTangibleHrs` to a number between 40 and 50 hours. 
5. do `npm install` `npm run build` and `npm start` to run this PR locally
6. After 1 minute, check that 50 hours for 1 week is not assigned. (you can do this by checking the database or by frontend)
7. Then edit the `lastWeekTangibleHrs` to a number between 50 - 60 hours, then check if 50 hours for 1 week are assigned but 60 hours for 1 week is not. 

## Screenshots or videos of changes:
### First video
https://github.com/OneCommunityGlobal/HGNRest/assets/113070229/aa7e1a8c-e5b4-48cf-b35c-4487ba012f15

### Second video
https://github.com/OneCommunityGlobal/HGNRest/assets/113070229/d731cfd2-cb5c-4cdf-885d-c6653f763f76


## Note:
